### PR TITLE
[nginx-ingress] set default ingress class to nginx

### LINKED ISSUE #4799: [ingress-nginx] Make "nginx" ingress class the default
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

Not all users know what is ingress class and why they should care. Also, users tend to create single controller. But everybody need to write something in the corresponding field in ingress controller manifest.  

Here where all ingress classes are caught and created
https://github.com/deckhouse/deckhouse/blob/main/modules/402-ingress-nginx/templates/controller/ingress-class.yaml#L10-L12 

### Proposed Solution

Make `spec.ingressClass` "nginx" by default

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by miklezzzz**
## Description
Having required to specify the .spec.ingressClass when defining an ingress nginx controller might seem daunting for some users. We can provide default value "nginx" if the field is left blank.

## Why do we need it, and what problem does it solve?
It makes it a bit easier to define an ingress nginx controller.
Close #4799 

## What is the expected result?
In case the .ingressClass isn't provided on create/update, we set it to 'nginx' via crd default field.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: ingress-nginx
type: feature
summary: Set ingressClass to 'nginx' if not explicitly set.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @miklezzzz. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by miklezzzz** [comment]
image

File: modules/402-ingress-nginx/crds/ingress-nginx.yaml [hunk 1]
```diff
@@ -35,6 +35,7 @@ spec:
 
                     **Caution!** If you set it to "nginx", then Ingress resources lacking the `kubernetes.io/ingress.class` annotation or `spec.ingressClassName` field will also be handled.
                   example: 'nginx'
+                  default: 'nginx'
                   pattern: '^[a-z0-9]([-a-z0-9]*[a-z0-9])?$'
```
Comment (yalosev) [inline modules/402-ingress-nginx/crds/ingress-nginx.yaml:39]: I guess we also have to change pattern here, according to https://github.com/deckhouse/deckhouse/pull/4932/

**Review by yalosev — CHANGES_REQUESTED** [review]
pattern change required

File: modules/402-ingress-nginx/crds/ingress-nginx.yaml (hunk 1 above)
Comment (miklezzzz) [reply modules/402-ingress-nginx/crds/ingress-nginx.yaml]: done, also updated secret name pattern with the same value

**Review by yalosev — APPROVED** [review]

**Review by nabokihms — APPROVED** [review]

**Pull request merged**